bricks.js

```diff
@@ -115,6 +115,46 @@ class Brick { // type 0 = invis, 1 = brick, 2 = question, 3 = block
             ctx.strokeRect(this.bottomBB.x - this.game.camera.x, this.bottomBB.y, this.bottomBB.width, this.bottomBB.height);
         }
     };
+
+    explode(){
+        this.game.addEntity(new Shard(this.game, this.x, this.y, -150));
+        this.game.addEntity(new Shard(this.game, this.x, this.y  - PARAMS.BLOCKWIDTH * 1.5, -150));
+        this.game.addEntity(new Shard(this.game, this.x + PARAMS.BLOCKWIDTH, this.y - PARAMS.BLOCKWIDTH * 1.5, 150));
+        this.game.addEntity(new Shard(this.game, this.x + PARAMS.BLOCKWIDTH, this.y, 150));
+        this.removeFromWorld = true;
+    }
+};
+
+class Shard{
+    constructor(game, x, y, xVelocity, yVelocity = -100){
+        Object.assign(this, {game, x, y})
+
+        this.animation = new Animator(ASSET_MANAGER.getAsset("./sprites/bricks.png"), 16, 0, 8, 8, 1, 0.33, 0, false, true);
+        this.velocity = {x:xVelocity, y:yVelocity};
+        this.life = 90; //90 clock ticks -> 1.5 seconds at 60 fps
+
+    }
+
+    update(){
+        const FALL_ACC = 562.5;
+        this.life -= 1;
+        if(this.life <= 0){
+            this.removeFromWorld = true;
+        }
+        this.velocity.y += FALL_ACC * this.game.clockTick;
+        this.y += this.game.clockTick * this.velocity.y * PARAMS.SCALE;
+        this.x += this.game.clockTick * this.velocity.x;
+    }
+
+    draw(ctx) {
+        this.animation.drawFrame(this.game.clockTick, ctx, this.x - this.game.camera.x, this.y, PARAMS.SCALE);
+    };
+
+    drawMinimap(ctx, mmX, mmY) {
+        ctx.fillStyle = "Brown";
+        ctx.fillRect(mmX + this.x / PARAMS.BITWIDTH, mmY + this.y / PARAMS.BITWIDTH, this.w / PARAMS.BITWIDTH, PARAMS.SCALE);
```

Comment:
In my personal option the shards probably shouldn't be drawn to the minimap because they don't effect the player in any way.